app/models/user.rb

```diff
@@ -419,17 +419,20 @@ def customize_digest(type)
 
   def generate_token
     user_id_and_time = { :id => id, :timestamp => Time.now }
-    encrypt(user_id_and_time)
+    User.encrypt(user_id_and_time)
   end
 
-  def validate_token(token)
-    decrypted_data = decrypt(token)
-    if id != decrypted_data[:id]
-      return false
-    elsif (Time.now - decrypted_data[:timestamp]) / 1.hour > 24.0
-      return false
+  def self.validate_token(token)
+    begin
+      decrypted_data = User.decrypt(token)      
+    rescue StandardError => e
```

Comment:
Instead of StandardError, can you please check specific type of exception it throws and catch only that.